spec/lib/audit/moab_to_catalog_spec.rb

```diff
@@ -123,38 +123,18 @@
         end
       end
 
-      context 'object does not exist' do
-        it 'calls #create when expect_to_create is true' do
-          expected_argument_list.each do |arg_hash|
-            expect(PreservedObject).to receive(:exists?).with(druid: arg_hash[:druid]).and_return(false)
-            exp_msg = "druid: #{arg_hash[:druid]} expected to exist in catalog but was not found"
-            expect(Rails.logger).to receive(:error).with(exp_msg)
-            expect(arg_hash[:po_handler]).to receive(:create)
-          end
-          described_class.check_existence_for_dir(storage_dir, true)
-        end
-        it 'does not call #create when expect_to_create is false' do
-          expected_argument_list.each do |arg_hash|
-            expect(PreservedObject).to receive(:exists?).with(druid: arg_hash[:druid]).and_return(false)
-            exp_msg = "druid: #{arg_hash[:druid]} expected to exist in catalog but was not found"
-            expect(Rails.logger).to receive(:error).with(exp_msg)
-            expect(arg_hash[:po_handler]).not_to receive(:create)
-          end
-          described_class.check_existence_for_dir(storage_dir) # expect_to_create is false by default
-        end
-      end
-
-      it "calls #confirm_version if object exists" do
+      it 'calls check_existence' do
+        fake_codes = %w[fake_code1 fake_code2]
         expected_argument_list.each do |arg_hash|
-          expect(PreservedObject).to receive(:exists?).with(druid: arg_hash[:druid]).and_return(true)
-          expect(arg_hash[:po_handler]).to receive(:confirm_version)
+          expect(arg_hash[:po_handler]).to receive(:check_existence).and_return(fake_codes)
         end
-        subject
+        # * 3 will magically give us a flat, 6 element array
+        expect(subject).to eq(fake_codes * 3)
```

Comment:
seems like this and line 137 are kind of doing the same thing?
ok to keep it.. Why does subject give us ` ["fake_code1", "fake_code2", "fake_code1", "fake_code2", "fake_code1", "fake_code2"] `? maybe im not understanding this test?

Reply:
Is it to show that `check_existence` returns an array of codes?